src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_TABLE_INCREASING = 2;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K,V> possibleNode = table[calculateIndex(key)];
+        if (possibleNode == null) {
+            table[calculateIndex(key)] = new Node<>(key, value);
```

Comment:
save index in variable not call method twice

Reply:
done